lib/data-types.js

```diff
@@ -441,16 +441,19 @@ class DATE extends ABSTRACT {
     return value;
   }
   _isChanged(value, originalValue) {
```

Comment:
I like that you wanted to refactor this `_isChanged` method, however I think an even better refactor can be done:

```js
_isChanged(value, originalValue) {
  if (value instanceof Date && originalValue instanceof Date) {
    return value.getTime() !== originalValue.getTime();
  }
  return value !== originalValue;
}
```

Please convince yourself that my code above is equivalent to the current code and use it instead :)